### PR TITLE
Add Tensor Board Connection

### PR DESCRIPTION
**Pull request by ahkarami**
Add Tensor Board Connection (for better visualization of training process)
Note that: Tensorboard & TensorFlow should be installed previously.

### CODE REVIEW

File: logger.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import scipy.misc
+import numpy as np
+import tensorflow as tf
+
+
+try:
+    from StringIO import StringIO  # Python 2.7
+except ImportError:
+    from io import BytesIO  # Python 3.x
+
```
Comment (louis-she) [inline logger.py]: no need to support python2.7, we have announced that in `README.md`

File: trainer.py [hunk 2]
```diff
@@ -29,6 +30,15 @@ def __init__(self, optimizer, model, training_dataloader,
         self.max_epoch = max_epoch
         self.persist_stride = persist_stride
 
+        # Set the logger for TensorBoard:
+        if Config.TENSOR_BOARD:
+            self.logger = Logger(Config.tensorBoardLoggerAddress)
+
+        # Make Tensor Board Log Directory (if not exists):
+        if Config.TENSOR_BOARD:
```
Comment (louis-she) [inline trainer.py]: merge the two `if` block.

File: trainer.py [hunk 3]
```diff
@@ -187,27 +197,67 @@ def run_epoch(self, mode):
...
+                'average_validation_loss_regression': total_reg_loss / total_iter,
+                'average_validation_total_loss': total_loss / total_iter,
+            }
+
+            for tag, value in info.items():
+                step = self.current_epoch
+                self.logger.scalar_summary(tag, value, step)
+
+
+def persist(self, is_best=False):
```
Comment (louis-she) [inline trainer.py]: the indentation of this function is not right, it this function is part of the class.

**Comment by louis-she** [comment]
Great work @ahkarami ,
Please see the comment I leave and fix them, I will test the function in this week.

**Comment by ahkarami** [comment]
Dear @louis-she,
I have changed & fix all of your requested issues. Know, i think you can merge the PR in the master.

**Comment by louis-she** [comment]
Cool @ahkarami ,
I will merge this branch after testing it, it will be in this week.

**Pull request merged**